packages/aws-cdk-lib/aws-logs/lib/subscription-filter.ts

```diff
@@ -62,6 +64,7 @@ export class SubscriptionFilter extends Resource {
       destinationArn: destProps.arn,
       roleArn: destProps.role && destProps.role.roleArn,
       filterPattern: props.filterPattern.logPatternString,
+      filterName: this.physicalName,
```

Comment:
wait, won't this change all subscription filters even if they don't supply `props.filterName`? Why is there a need to propagate the `filterName` to `physicalName` this way

Reply:
Thank you for your comment!

---

> won't this change all subscription filters even if they don't supply props.filterName? 

In my understanding, if we don't specify `filterName` prop introduced by this PR, `this.physicalName` will be undefined and then `filterName` for L1 CfnSubscriptionFilter construct will be also undefined. As a result, FilterName porperty for [AWS::Logs::SubscriptionFilter](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-logs-subscriptionfilter.html) resource in generted CFn template will be omitted. I already tested with the sample app in my env. 

---

> Why is there a need to propagate the filterName to physicalName this way

I align this with the similar constructs such as LogGroup, LogStream or MetricFilter as below.

https://github.com/aws/aws-cdk/blob/51b20f9d07f66ddf677cea62c75c13f20b066293/packages/aws-cdk-lib/aws-logs/lib/log-group.ts#L466-L469
https://github.com/aws/aws-cdk/blob/9a3399f11c18cb8c2f5a92dd3a6773681aa77e39/packages/aws-cdk-lib/aws-logs/lib/log-stream.ts#L67-L70
https://github.com/aws/aws-cdk/blob/9a3399f11c18cb8c2f5a92dd3a6773681aa77e39/packages/aws-cdk-lib/aws-logs/lib/metric-filter.ts#L25-L28

From CFn perspective, if we specify `FilterName` property, this is used as physical id. Shouldn't we propagate the filterName to physicalName?

Reply:
We can't, because that would be a breaking change for users who don't specify `filterName`. The behavior that existed before needs to be maintained, unfortunately.

In practice, if we really want to change the behavior we hide it behind a feature flag. But this ins't important enough to go down that route, imo.

Reply:
Oh wait you're saying `this.physicalName` will be undefined if no `filterName` is given... hmm maybe thats the case but I would be surprised.

Reply:
Nevermind, you are right :)